aspnetcore/grpc/retries.md

```diff
@@ -0,0 +1,137 @@
+    RetryPolicy = new RetryPolicy
+    {
+        MaxAttempts = 5,
+        InitialBackoff = TimeSpan.FromSeconds(1),
+        MaxBackoff = TimeSpan.FromSeconds(5),
+        BackoffMultiplier = 1.5,
+        RetryableStatusCodes = { StatusCode.Unavailable }
+    }
+};
+
+var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
+{
+    ServiceConfig = new ServiceConfig { MethodConfigs = { defaultMethodConfig } }
+});
+```
+
+The preceding code:
+
+1. Creates a `MethodConfig`. Retry policies can be configured per-method and methods are matched using the `Names` property. This method is configured with `MethodName.Default` so it is applied to all gRPC methods called by this channel.
+2. Configures a retry policy. This policy instructs clients to automatically retry gRPC calls that fail with the status code `Unavailable`.
+3. Configures the created channel to use the retry policy by setting `GrpcChannelOptions.ServiceConfig`.
+
+gRPC clients created with the channel will automatically retry failed calls:
+
+```csharp
+var client = new Greeter.GreeterClient(channel);
+var response = await client.SayHelloAsync(
+    new HelloRequest { Name = ".NET" });
+
+Console.WriteLine("From server: " + response.Message);
+```
+
+### gRPC retry options
+
+The following table describes options for configuring gRPC retry policies:
+
+| Option | Description |
+| ------ | ----------- |
+| MaxAttempts | The maximum number of call attempts, including the original attempt. This value is limited by `GrpcChannelOptions.MaxRetryAttempts` which defaults to 5. A value is required and must be greater than 1. |
+| InitialBackoff | The initial backoff delay between retry attempts. A randomized delay between 0 and the current backoff will determine when the next retry attempt is made. After each attempt the current backoff is multiplied by `BackoffMultiplier`. A value is required and must be greater than zero. |
+| MaxBackoff | The maximum backoff places an upper limit on exponential backoff growth. Value is required and must be greater than zero. |
+| BackoffMultiplier | The backoff will be multiplied by this value after each retry attempt and will increase exponentially when the multiplier is greater than 1. A value is required and must be greater than 0. |
+| RetryableStatusCodes | A collection of status codes. A gRPC call that fails with a matching status will be automatically retried. For more information about status codes, see [Status codes and their use in gRPC](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). At least one status code is required. |
+
+## Hedging
+
+Hedging is an alternative retry strategy. Hedging enables aggressively sending multiple copies of a single gRPC call without waiting for a response. Because hedged gRPC calls may be executed multiple times on the server, it is important that hedging is only enabled for methods that are safe to execute multiple times without adverse affect.
```

Comment:
```suggestion
Hedging is an alternative retry strategy. Hedging enables aggressively sending multiple copies of a single gRPC call without waiting for a response. Hedged gRPC calls may be executed multiple times on the server. So it's important that hedging is only enabled for methods that are safe to execute multiple times without adverse affect.
```

Reply:
Starting a new sentence with "So it's..." seems strange to me.

Reply:
I removed the "So" at the beginning of the sentence.